File: languages/python/bloock/entity/encryption/rsa_encrypter.py
import bloock._bridge.proto.encryption_entities_pb2 as proto
from bloock.entity.encryption.encrypter import Encrypter
from bloock.entity.encryption.encrypter_args import EncrypterArgs


class RsaEncrypter(Encrypter):
    def __init__(self, public_key: str) -> None:
        super().__init__(alg=proto.RSA, args=EncrypterArgs(public_key))

    def to_proto(self) -> proto.Encrypter:
        return proto.Encrypter(alg=self.alg, args=self.args.to_proto())

File: languages/php/Bloock/Entity/Encryption/RsaEncrypter.php
<?php

namespace Bloock\Entity\Encryption;

class RsaEncrypter implements Encrypter
{
    public int $alg;
    public EncrypterArgs $args;

    public function __construct(EncrypterArgs $args)
    {
        $this->alg = \Bloock\EncryptionAlg::RSA;
        $this->args = $args;
    }

    public function toProto(): \Bloock\Encrypter
    {
        $p = new \Bloock\Encrypter();
        $p->setAlg($this->alg);

        if ($this->args->localKey != null) {
            $p->setLocalKey($this->args->localKey->toProto());
        }

        if ($this->args->managedKey != null) {
            $p->setManagedKey($this->args->managedKey->toProto());
        }

        return $p;
    }
}